lib/charms/opensearch/v0/opensearch_backups.py

```diff
@@ -387,94 +408,70 @@ def __init__(self, charm: "OpenSearchBaseCharm", relation_name: str = PeerCluste
             self.charm.on[S3_RELATION].relation_broken, self._on_s3_relation_broken
         )
 
+        for event in [
+            charm.on[PeerClusterRelationName].relation_joined,
+            charm.on[PeerClusterRelationName].relation_changed,
+            charm.on[PeerClusterRelationName].relation_departed,
+        ]:
+            # We need to keep track of the peer-cluster relation
+            # A unit-level secret will not trigger secret changes
+
+            # I've discussed this offline with @wallyworld and the main idea
+            # is that the unit is already aware of the secret change, why triggering
+            # a new hook in this case?
+            # Now, opensearch_backups.py was originally using opensearch_secrets.py to
+            # update its the s3-credentials secret and inform this class via "manual_update"
+
+            # Listening to the peer cluster relation is another alternative:
+            # Effectively it will call the common method that both _on_secret_changed and
+            # _on_peer_cluster_relation_event uses to update the keystore.
+            self.framework.observe(event, self._on_peer_cluster_relation_event)
```

Comment:
Sure, but my goal here is exactly to break the big peer-cluster handler out. That method is massive and I cannot really grasp what it does on each case. There is no real risk of *listening* to a complex event if we do not defer, as we are not necessarily making changes to its databag (hence, we are not *triggering* peer-cluster-changed).

Now, to your point, I noticed we defer.

So, I will first try by removing that defer.

The other option is to listen to this event, then we raise a custom event that is focused only on processing peer-cluster relations changes *for backups*. This way, we can defer this small event, internal, backup event instead of the peer cluster.